pkg/client/edv/client.go

```diff
@@ -97,9 +100,47 @@ func (c *Client) ReadDocument(vaultID, docID string) (*models.EncryptedDocument,
 	}
 }
 
+// QueryVault queries the given vault and returns the URLs of all documents that match the given query.
+func (c *Client) QueryVault(vaultID string, query *models.Query) ([]string, error) {
+	jsonToSend, err := c.marshal(query)
+	if err != nil {
+		return nil, err
+	}
+
+	// The linter falsely claims that the body is not being closed
+	// https://github.com/golangci/golangci-lint/issues/637
+	resp, err := c.httpClient.Post(fmt.Sprintf("%s/encrypted-data-vaults/%s/queries", //nolint: bodyclose
+		c.edvServerURL, url.PathEscape(vaultID)), "application/json", bytes.NewBuffer(jsonToSend))
+	if err != nil {
+		return nil, fmt.Errorf("failed to send POST message: %w", err)
+	}
+
+	defer closeReadCloser(resp.Body)
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response message while retrieving document: %w", err)
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		docURLs := make([]string, 0)
```

Comment:
You don't need to 'make' a 0 length array. Just declare it as: var docURLs []string

Reply:
Good point. Done.